server/filestore.go

```diff
@@ -2370,7 +2370,41 @@ func (fs *fileStore) Purge() uint64 {
 		cb(-int64(purged), -rbytes, 0, _EMPTY_)
 	}
 
-	return purged
+	return purged, nil
+}
+
+// Compact will remove all messages from this store up to
+// but not including the seq parameter.
+// Will return the number of purged messages.
+func (fs *fileStore) Compact(seq uint64) (uint64, error) {
+	if seq == 0 {
+		return fs.Purge()
+	}
+
+	if fs.isClosed() {
+		return 0, ErrStoreClosed
+	}
+
+	if _, err := fs.msgForSeq(seq); err != nil {
+		return 0, ErrStoreMsgNotFound
+	}
+
+	var purged uint64
+
+	for fseq := fs.firstSeq(); fseq < seq; fseq = fs.firstSeq() {
+		if found, _ := fs.removeMsg(fseq, false); found {
```

Comment:
Either rewrite this function with full lock or check error here since store may close. Also I would not recommend ignoring an error here since this may be critical info for caller.

Reply:
Fair. Will update.

Reply:
fixed.